pkgs/tools/system/osquery/default.nix

```diff
@@ -0,0 +1,82 @@
+{ pkgs }:
+
+with pkgs;
+let
+  buildStdenv = overrideCC stdenv llvmPackages.clangUseLLVM;
+  opensslArchive =
+    let
+      # https://github.com/osquery/osquery/blob/877d5e69ab97e15800b5c5128b3de2cf6f322942/libraries/cmake/formula/openssl/CMakeLists.txt#L3-L4.
+      version = "1.1.1q";
+      sha256 = "d7939ce614029cdff0b6c20f0e2e5703158a489a72b2507b8bd51bf8c8fd10ca";
+    in
+    fetchurl {
+      inherit sha256;
+      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+    };
+in
+buildStdenv.mkDerivation rec {
+  pname = "osquery";
+  version = "5.5.1";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+
+    fetchSubmodules = true;
+    sha256 = "sha256-Q6PQVnBjAjAlR725fyny+RhQFUNwxWGjLDuS5p9JKlU=";
+  };
+
+  patches = [
+    ./Remove-git-reset.patch
+    ./Use-locale.h-instead-of-removed-xlocale.h-header.patch
+    ./Remove-circular-definition-of-AUDIT_FILTER_EXCLUDE.patch
+    # For current state of compilation against glibc in the clangWithLLVM toolchain, refer to the upstream issue in https://github.com/osquery/osquery/issues/7823.
+    ./Remove-system-controls-table.patch
+  ];
+
+
+  buildInputs = [
+    llvmPackages.libunwind
+  ];
+  nativeBuildInputs = [
+    # > Git (>= 2.14.0), CMake (>= 3.21.4), Python 3 are required to build.
+    # > The rest of the dependencies are downloaded by CMake.
+    # https://osquery.readthedocs.io/en/latest/development/building/
+    cmake
+    git
+    python3
+
+    # https://github.com/osquery/osquery/blob/877d5e69ab97e15800b5c5128b3de2cf6f322942/libraries/cmake/formula/openssl/CMakeLists.txt#L94
+    perl
+  ];
+
+  # Fix absolute packaging paths.
+  prePatch = ''
+    substituteInPlace cmake/install_directives.cmake --replace "/control" "control"
+  '';
+
+  # For explanation of these deletions, refer to the ./Use-locale.h-instead-of-removed-xlocale.h-header.patch file.
+  preConfigure = ''
+    find libraries/cmake/source -name 'config.h' -exec sed -i '/#define HAVE_XLOCALE_H 1/d' {} \;
+  '';
+
+  cmakeFlags = [
+    "-DOSQUERY_VERSION=${version}"
+    "-DOSQUERY_OPENSSL_ARCHIVE_PATH=${opensslArchive}"
+  ];
+
+  postFixup = ''
+    patchelf --set-rpath "${lib.makeLibraryPath buildInputs }:$(patchelf --print-rpath $out/bin/osqueryd)" "$out/bin/osqueryd"
```

Comment:
there is a trailing space after buildInputs

Reply:
BTW, why is this needed at all? I think Nix should do it for you? Did you meant `propagatedBuildInputs` in fact somewhere?

Reply:
Unfortunately, even when I use `llvmPackage.libunwind` as a propagated build input, the shared library `libunwind.so.1` is not found.

Reply:
I agree it would be better to let the stdenv doing it for us. 
However @jdbaldry tried to do it and it didn't work out of the box. I don't think manually patching the binary is a big issue and we could improve this build later, without any breaking changes.
So, I would propose to tolerate this minor issue.

Reply:
Let's move on :)